common/src/main/java/com/microsoft/identity/common/internal/controllers/LocalMSALController.java

```diff
@@ -316,25 +317,49 @@ public AcquireTokenResult acquireTokenSilent(
         // subsequent CacheRecords represent other profiles (projections) of this principal in
         // other tenants. Those tokens will be 'sparse', meaning that their AT/RT will not be loaded
         final ICacheRecord fullCacheRecord = cacheRecords.get(0);
-
-        if (accessTokenIsNull(fullCacheRecord)
+        if (fullCacheRecord.getAccessToken().refreshOnIsActive()) {
+            Logger.info(
+                    TAG,
+                    "RefreshIn is active. This will extend your token usage in the rare case servers are not available."
+            );
+        }
+        if (parameters.isRefreshIn()) { //refresh_in second call
+            AccessTokenRecord accessTokenRecord = fullCacheRecord.getAccessToken();
+            Logger.warn(
+                    TAG + methodName,
+                    "Attempting renewal of Access Token because it's refresh-expired. RefreshIn was expired at " + accessTokenRecord.getRefreshOn() + ". Regular expiry is at " + accessTokenRecord.getExpiresOn() + "."
+                            + "Currently executing initial command, SilentTokenCommand with CorrelationId: " + parameters.getCorrelationId() + " subsequent RefreshCommand with CorrelationId not yet available. See further in logs."
```

Comment:
initial command?

Reply:
We don't have to.. but, we're also not just telling them what we're about to do in the future, we're additionally providing when the AT os set to refresh-expire and expire.

Reply:
Outdated